### DISCUSSION TITLE
Changelog picking up wrong username

### DISCUSSION THREAD

**Discussion by manikmagar** [jreleaser/jreleaser#1111] [General]
I am attempting to use JReleaser for one internal tool. The early snapshot version got properly created but change log's contributor section picked a wrong github username. 

Looks like it is searching by Author's name - 'Kevin King' (see web query [here](https://github.com/search?l=&p=7&q=fullname%3AKevin+King+type%3Auser&type=Users)) and just picked up the first user found 
https://github.com/jreleaser/jreleaser/blob/main/sdks/jreleaser-github-java-sdk/src/main/java/org/jreleaser/sdk/github/XGithub.java#L103-L111 - 

But that's not the Kevin who committed the change :). This user also has [a private github email](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#:~:text=GitHub%20account.%22-,Note%3A,-If%20you%20created) id - **ID+USERNAME@users.noreply.github.com**. The API documentation states that users having "Public email" listed on their profile can only be searched using email and since there is no public email listed here, search by email id is not useful. Searching by names as seen below can result in unexpected results. I can't reproduce it with my name since there is just one Manik Magar on github, technically two profiles but both me (strange, right! 😄 ) 

For users with Github private email address, I wonder if we can extract username from email id and use that as a fallback before searching by name. I guess that would give better/accurate user lookup than search by name in such cases.

**Comment by aalmiray** [comment]
And how would be able to tell that the first user found with a matching name is not the real user you'd like to see in the changelog?

I can see the fallback being: if no matching email nor name then the email's username is used.

**Reply by manikmagar** [reply]
> > And how would be able to tell that the first user found with a matching name is not the real user you'd like to see in the changelog?

No no, I don't mean JReleaser to do that validation at all. 

Here is what I am thinking - 

1. Search by email.
2. If user not found and email is a private email id then extract username from it and use that.
3. If email is not a private one, lookup by name.

I hope that makes sense. Search by name is the last thing to do, it is not reliable IMO.

**Reply by aalmiray** [reply]
Ah gotcha. I get it know. I guess it can be short circuit to

1. check email, if private then extract username. done.
2. search by email. If found use name.
3. use name as extracted from commit message (this would need to be recorded as well. I think only the email is recorded at the moment.

How about that?

**Reply by manikmagar** [reply]
Yeah, that sounds better and efficient search. I think RevCommit does include name and it is recorded already (if I got it right).

**Reply by aalmiray** [reply]
Indeed. `RevCommit` provides author information. However, the name may also be specified in a footer such as "Co-authored-by".

**Reply by manikmagar** [reply]
Yes, thats correct. Should I create an issue from this?

**Reply by aalmiray** [reply]
Yes, please, that would be really appreciated 😄

**Reply by manikmagar** [reply]
Here it is https://github.com/jreleaser/jreleaser/issues/1112.

**Reply by manikmagar** [reply]
and if its okay then I might give it a try later today or tomorrow, unless its done already ;)

**Reply by aalmiray** [reply]
Sure, feel free to take it 😅

**Reply by manikmagar** [reply]
Fixed with https://github.com/jreleaser/jreleaser/issues/1112
